pandas/core/frame.py

```diff
@@ -3973,10 +3973,25 @@ def set_index(self, keys, drop=True, append=False, inplace=False,
         -------
         dataframe : DataFrame
         """
-        inplace = validate_bool_kwarg(inplace, 'inplace')
+        from pandas import Series
+
         if not isinstance(keys, list):
             keys = [keys]
 
+        col_labels = [x for x in keys
+                      if not isinstance(x, (Series, Index, MultiIndex,
+                                            list, np.ndarray))]
+        if any(x not in self for x in col_labels):
+            missing = [x for x in col_labels if x not in self]
```

Comment:
are there explict tests for each of the branches. pls add a comment to each one indicating what you are checking.